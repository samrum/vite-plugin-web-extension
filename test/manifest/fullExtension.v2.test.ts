import { getResourceDir, runManifestV2Test } from "./manifestTestUtils";

const resourceDir = getResourceDir("fullExtension");

runManifestV2Test("fullExtension", () => ({
  background: {
    scripts: [`${resourceDir}/src/entries/background/main.js`],
    persistent: false,
  },
  content_scripts: [
    {
      js: [`${resourceDir}/src/entries/contentScript/primary/main.js`],
      matches: ["*://*/*"],
    },
  ],
  web_accessible_resources: [`${resourceDir}/src/lib.js`],
  browser_action: {
    default_icon: {
      32: `${resourceDir}/src/assets/logo.svg`,
    },
    // default_popup: `${resourceDir}/src/entries/popup/index.html`,
  },
  options_ui: {
    chrome_style: false,
    open_in_tab: true,
    page: `${resourceDir}/src/entries/options/index.html`,
  },
  permissions: ["*://*/*"],
}));
