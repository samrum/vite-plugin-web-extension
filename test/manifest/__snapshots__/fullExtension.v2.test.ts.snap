// Vitest Snapshot v1, https://vitest.dev/guide/snapshot.html

exports[`fullExtension - Manifest V2 1`] = `
[
  {
    "code": "browser.runtime.onInstalled.addListener(() => {
  console.log("Extension installed");
});
browser.action.onClicked.addListener(async () => {
  const tab = await getCurrentTab();
  chrome.scripting.executeScript({
    target: { tabId: tab == null ? void 0 : tab.id },
    files: ["src/lib.js"]
  });
});
async function getCurrentTab() {
  let queryOptions = { active: true, lastFocusedWindow: true };
  let [tab] = await chrome.tabs.query(queryOptions);
  return tab;
}
",
    "dynamicImports": [],
    "exports": [],
    "facadeModuleId": "vite-plugin-web-extension/background.html",
    "fileName": "assets/background.js",
    "implicitlyLoadedBefore": [],
    "importedBindings": {},
    "imports": [],
    "isDynamicEntry": false,
    "isEntry": true,
    "isImplicitEntry": false,
    "map": null,
    "modules": {
      "vite-plugin-web-extension/background.html": {
        "code": "",
        "originalLength": 182,
        "removedExports": [],
        "renderedExports": [],
        "renderedLength": 0,
      },
      "vite-plugin-web-extension/test/manifest/resources/fullExtension/src/entries/background/main.js": {
        "code": "browser.runtime.onInstalled.addListener(() => {
  console.log("Extension installed");
});

browser.action.onClicked.addListener(async () => {
  const tab = await getCurrentTab();

  chrome.scripting.executeScript({
    target: { tabId: tab?.id },
    files: ["src/lib.js"],
  });
});

async function getCurrentTab() {
  let queryOptions = { active: true, lastFocusedWindow: true };
  let [tab] = await chrome.tabs.query(queryOptions);
  return tab;
}",
        "originalLength": 451,
        "removedExports": [],
        "renderedExports": [],
        "renderedLength": 450,
      },
    },
    "name": "background",
    "referencedFiles": [],
    "type": "chunk",
    "viteMetadata": {
      "importedAssets": Set {},
      "importedCss": Set {},
    },
  },
  {
    "fileName": "assets/index.css",
    "name": "index.css",
    "source": "#app {
  font-family: Avenir, Helvetica, Arial, sans-serif;
  -webkit-font-smoothing: antialiased;
  -moz-osx-font-smoothing: grayscale;
  text-align: center;
  color: #2c3e50;
}

h1 {
  margin: 5px;
}
",
    "type": "asset",
  },
  {
    "code": "const logo = "data:image/svg+xml,%3c?xml%20version='1.0'?%3e%3csvg%20width='443'%20height='681'%20xmlns='http://www.w3.org/2000/svg'%20xmlns:svg='http://www.w3.org/2000/svg'%3e%3cg%20class='layer'%3e%3ctitle%3eLayer%201%3c/title%3e%3cpath%20d='m0.075734,0c9,26.55%2018.84,52.81%2028.24,79.22c30.33,84.6%2061.3,168.97%2091.22,253.72c13.25,36.73%2026.29,73.53%2039.63,110.23c-10.08,18.43%20-21.82,35.89%20-32.39,54.04c-11.44,18.89%20-23.09,37.65%20-34.35,56.64c-5.33,-4.89%20-9.8,-10.61%20-14.54,-16.05c-12.73,-14.88%20-26.15,-29.16%20-38.82,-44.09c-6.28,-7.08%20-12.43,-14.29%20-18.93,-21.18c-1.49,-1.39%20-2.02,-3.38%20-1.93,-5.36c-0.14,-12.72%20-1.21,-25.4%20-1.33,-38.13c-1.05,-10.52%20-0.41,-21.13%20-1.47,-31.64c-1.66,-49.12%20-4,-98.2%20-5.66,-147.32c-0.83,-7.97%20-0.45,-16%20-1.05,-23.98c-0.83,-9.03%20-0.36,-18.13%20-1.09,-27.17c-0.71,-9.28%20-0.3,-18.61%20-1.21,-27.88c-0.41,-8.08%20-0.15,-16.19%20-0.95,-24.24c-0.47,-7.29%20-0.19,-14.6%20-0.85,-21.88c-0.6,-9.62%20-0.33,-19.28%20-1.15,-28.89c-0.43,-8.42%20-0.2,-16.87%20-1.08,-25.27c-0.33,-6.55%20-0.2,-13.12%20-0.66,-19.67c-0.79,-9.63%20-0.37,-19.32%20-1.25,-28.95c-0.16,-7.38%20-0.64,-14.77%20-0.38,-22.15z'%20fill='%23c72a21'%20id='svg_3'/%3e%3cpath%20d='m440.685734,8.63c0.57,-1.69%201.42,-3.22%202.55,-4.6c0.24,11.68%20-0.29,23.38%20-0.94,35.04c-0.73,30.7%20-2.75,61.33%20-3.5,92.02c-0.98,9.85%20-0.42,19.77%20-1.37,29.62c-3.25,81.67%20-6.64,163.35%20-9.83,245.03c-0.8,8.77%20-0.3,17.6%20-1.23,26.36c-0.38,12.82%20-0.86,25.64%20-1.55,38.43c-2.18,3.34%20-5.12,6.1%20-7.75,9.08c-10.72,11.63%20-20.63,23.98%20-31.42,35.53c-11.53,12.89%20-22.72,26.06%20-34.24,38.95c-3.65,-4.46%20-6.13,-9.71%20-9.17,-14.58c-18.5,-30.79%20-37.34,-61.38%20-55.82,-92.19c-1.32,-2%20-2.5,-4.41%20-1.19,-6.73c29.27,-80.86%2057.94,-161.94%2087.23,-242.8c22.91,-62.99%2045.73,-126.02%2068.23,-189.16z'%20fill='%23c72a21'%20id='svg_4'/%3e%3cpath%20d='m191.645734,154.35c10.01,8.5%2019.91,17.15%2029.98,25.58c10.51,-8.2%2020.23,-17.36%2030.65,-25.66c-8.56,19.24%20-17.64,38.23%20-26.13,57.5c-1.38,3.04%20-2.79,6.07%20-4.39,9c-4.36,-8.47%20-7.84,-17.36%20-11.94,-25.95c-6.05,-13.49%20-12.55,-26.79%20-18.17,-40.47z'%20fill='%23c72a21'%20id='svg_5'/%3e%3cpath%20d='m221.715734,242.6c8.3,-16.28%2015.29,-33.2%2023.34,-49.6c1.29,7.8%200.66,15.73%201.66,23.56c5.3,91.85%2010.58,183.69%2015.97,275.54c3.53,62.97%207.56,125.91%2010.71,188.9c-17.5,-19.12%20-34.25,-38.95%20-51.8,-58.02c-17.24,19.36%20-34.48,38.73%20-51.63,58.16c-0.26,-9.38%201.2,-18.68%201.27,-28.04c9.08,-153.04%2017.9,-306.09%2027,-459.14c0.38,0.36%201.14,1.08%201.52,1.44c7.33,15.72%2014.22,31.67%2021.96,47.2z'%20fill='%23c72a21'%20id='svg_6'/%3e%3cpath%20d='m272.365734,444.25c3.62,4.28%205.9,9.46%208.85,14.2c19.88,32.82%2039.74,65.65%2059.4,98.6c3.73,6.33%207.88,12.43%2011.13,19.03c-8.54,14.4%20-17.82,28.34%20-26.67,42.55c-10.55,17.39%20-21.78,34.35%20-32.4,51.71c-2.44,3.3%20-3.93,7.39%20-7.22,10c-4.12,-70.18%20-7.81,-140.37%20-11.87,-210.55c-0.31,-8.52%20-1.29,-17.02%20-1.22,-25.54z'%20fill='%23c72a21'%20id='svg_7'/%3e%3cpath%20d='m162.075734,458.55c3.04,-4.63%205.04,-9.98%208.96,-13.98c-0.38,15.47%20-1.74,30.89%20-2.45,46.35c-3.51,63.44%20-7.32,126.85%20-10.68,190.29c-10.54,-15.71%20-20.13,-32.05%20-30.49,-47.88c-11.75,-19.27%20-24.03,-38.19%20-35.88,-57.39c2.35,-5.1%205.57,-9.72%208.46,-14.52c20.49,-34.42%2041.32,-68.62%2062.08,-102.87z'%20fill='%23c72a21'%20id='svg_8'/%3e%3c/g%3e%3c/svg%3e";
export {
  logo as l
};
",
    "dynamicImports": [],
    "exports": [
      "l",
    ],
    "facadeModuleId": null,
    "fileName": "assets/logo.js",
    "implicitlyLoadedBefore": [],
    "importedBindings": {},
    "imports": [],
    "isDynamicEntry": false,
    "isEntry": false,
    "isImplicitEntry": false,
    "map": null,
    "modules": {
      "vite-plugin-web-extension/test/manifest/resources/fullExtension/src/assets/logo.svg": {
        "code": "const logo = "data:image/svg+xml,%3c?xml%20version='1.0'?%3e%3csvg%20width='443'%20height='681'%20xmlns='http://www.w3.org/2000/svg'%20xmlns:svg='http://www.w3.org/2000/svg'%3e%3cg%20class='layer'%3e%3ctitle%3eLayer%201%3c/title%3e%3cpath%20d='m0.075734,0c9,26.55%2018.84,52.81%2028.24,79.22c30.33,84.6%2061.3,168.97%2091.22,253.72c13.25,36.73%2026.29,73.53%2039.63,110.23c-10.08,18.43%20-21.82,35.89%20-32.39,54.04c-11.44,18.89%20-23.09,37.65%20-34.35,56.64c-5.33,-4.89%20-9.8,-10.61%20-14.54,-16.05c-12.73,-14.88%20-26.15,-29.16%20-38.82,-44.09c-6.28,-7.08%20-12.43,-14.29%20-18.93,-21.18c-1.49,-1.39%20-2.02,-3.38%20-1.93,-5.36c-0.14,-12.72%20-1.21,-25.4%20-1.33,-38.13c-1.05,-10.52%20-0.41,-21.13%20-1.47,-31.64c-1.66,-49.12%20-4,-98.2%20-5.66,-147.32c-0.83,-7.97%20-0.45,-16%20-1.05,-23.98c-0.83,-9.03%20-0.36,-18.13%20-1.09,-27.17c-0.71,-9.28%20-0.3,-18.61%20-1.21,-27.88c-0.41,-8.08%20-0.15,-16.19%20-0.95,-24.24c-0.47,-7.29%20-0.19,-14.6%20-0.85,-21.88c-0.6,-9.62%20-0.33,-19.28%20-1.15,-28.89c-0.43,-8.42%20-0.2,-16.87%20-1.08,-25.27c-0.33,-6.55%20-0.2,-13.12%20-0.66,-19.67c-0.79,-9.63%20-0.37,-19.32%20-1.25,-28.95c-0.16,-7.38%20-0.64,-14.77%20-0.38,-22.15z'%20fill='%23c72a21'%20id='svg_3'/%3e%3cpath%20d='m440.685734,8.63c0.57,-1.69%201.42,-3.22%202.55,-4.6c0.24,11.68%20-0.29,23.38%20-0.94,35.04c-0.73,30.7%20-2.75,61.33%20-3.5,92.02c-0.98,9.85%20-0.42,19.77%20-1.37,29.62c-3.25,81.67%20-6.64,163.35%20-9.83,245.03c-0.8,8.77%20-0.3,17.6%20-1.23,26.36c-0.38,12.82%20-0.86,25.64%20-1.55,38.43c-2.18,3.34%20-5.12,6.1%20-7.75,9.08c-10.72,11.63%20-20.63,23.98%20-31.42,35.53c-11.53,12.89%20-22.72,26.06%20-34.24,38.95c-3.65,-4.46%20-6.13,-9.71%20-9.17,-14.58c-18.5,-30.79%20-37.34,-61.38%20-55.82,-92.19c-1.32,-2%20-2.5,-4.41%20-1.19,-6.73c29.27,-80.86%2057.94,-161.94%2087.23,-242.8c22.91,-62.99%2045.73,-126.02%2068.23,-189.16z'%20fill='%23c72a21'%20id='svg_4'/%3e%3cpath%20d='m191.645734,154.35c10.01,8.5%2019.91,17.15%2029.98,25.58c10.51,-8.2%2020.23,-17.36%2030.65,-25.66c-8.56,19.24%20-17.64,38.23%20-26.13,57.5c-1.38,3.04%20-2.79,6.07%20-4.39,9c-4.36,-8.47%20-7.84,-17.36%20-11.94,-25.95c-6.05,-13.49%20-12.55,-26.79%20-18.17,-40.47z'%20fill='%23c72a21'%20id='svg_5'/%3e%3cpath%20d='m221.715734,242.6c8.3,-16.28%2015.29,-33.2%2023.34,-49.6c1.29,7.8%200.66,15.73%201.66,23.56c5.3,91.85%2010.58,183.69%2015.97,275.54c3.53,62.97%207.56,125.91%2010.71,188.9c-17.5,-19.12%20-34.25,-38.95%20-51.8,-58.02c-17.24,19.36%20-34.48,38.73%20-51.63,58.16c-0.26,-9.38%201.2,-18.68%201.27,-28.04c9.08,-153.04%2017.9,-306.09%2027,-459.14c0.38,0.36%201.14,1.08%201.52,1.44c7.33,15.72%2014.22,31.67%2021.96,47.2z'%20fill='%23c72a21'%20id='svg_6'/%3e%3cpath%20d='m272.365734,444.25c3.62,4.28%205.9,9.46%208.85,14.2c19.88,32.82%2039.74,65.65%2059.4,98.6c3.73,6.33%207.88,12.43%2011.13,19.03c-8.54,14.4%20-17.82,28.34%20-26.67,42.55c-10.55,17.39%20-21.78,34.35%20-32.4,51.71c-2.44,3.3%20-3.93,7.39%20-7.22,10c-4.12,-70.18%20-7.81,-140.37%20-11.87,-210.55c-0.31,-8.52%20-1.29,-17.02%20-1.22,-25.54z'%20fill='%23c72a21'%20id='svg_7'/%3e%3cpath%20d='m162.075734,458.55c3.04,-4.63%205.04,-9.98%208.96,-13.98c-0.38,15.47%20-1.74,30.89%20-2.45,46.35c-3.51,63.44%20-7.32,126.85%20-10.68,190.29c-10.54,-15.71%20-20.13,-32.05%20-30.49,-47.88c-11.75,-19.27%20-24.03,-38.19%20-35.88,-57.39c2.35,-5.1%205.57,-9.72%208.46,-14.52c20.49,-34.42%2041.32,-68.62%2062.08,-102.87z'%20fill='%23c72a21'%20id='svg_8'/%3e%3c/g%3e%3c/svg%3e";",
        "originalLength": 3412,
        "removedExports": [],
        "renderedExports": [
          "default",
        ],
        "renderedLength": 3411,
      },
    },
    "name": "logo",
    "referencedFiles": [],
    "type": "chunk",
    "viteMetadata": {
      "importedAssets": Set {},
      "importedCss": Set {},
    },
  },
  {
    "fileName": "assets/main.css",
    "name": "main.css",
    "source": ".logo {
  z-index: 99999;
  position: fixed;
  bottom: 20px;
  right: 10px;
  width: 60px;
  height: 60px;
  display: flex;
  justify-content: center;
  align-items: center;
  border: 4px solid #c72a21;
  border-radius: 50%;
  background-color: #fff;
}

img {
  position: absolute;
  top: 7px;
}
",
    "type": "asset",
  },
  {
    "code": "import { l as logo } from "../../../../../../../../logo.js";
async function renderContent(cssPaths, render = (_appRoot) => {
}) {
  console.log("renderContent", cssPaths);
}
renderContent(["assets/main.css"], (appRoot) => {
  const logoImageUrl = new URL(logo, import.meta.url).href;
  appRoot.innerHTML = \`
    <div class="logo">
      <img src="\${logoImageUrl}" height="50" alt="" />
    </div>
  \`;
});
console.log(chrome.runtime.getURL("src/lib.js"));
",
    "dynamicImports": [],
    "exports": [],
    "facadeModuleId": "vite-plugin-web-extension/test/manifest/resources/fullExtension/src/entries/contentScript/primary/main.js",
    "fileName": "assets/test/manifest/resources/fullExtension/src/entries/contentScript/primary/main.js",
    "implicitlyLoadedBefore": [],
    "importedBindings": {
      "assets/logo.js": [
        "l",
      ],
    },
    "imports": [
      "assets/logo.js",
    ],
    "isDynamicEntry": false,
    "isEntry": true,
    "isImplicitEntry": false,
    "map": null,
    "modules": {
      "vite-plugin-web-extension/test/manifest/resources/fullExtension/src/entries/contentScript/primary/main.js": {
        "code": "renderContent(import.meta.PLUGIN_WEB_EXT_CHUNK_CSS_PATHS, (appRoot) => {
  const logoImageUrl = new URL(logo, import.meta.url).href;

  appRoot.innerHTML = \`
    <div class="logo">
      <img src="\${logoImageUrl}" height="50" alt="" />
    </div>
  \`;
});

console.log(chrome.runtime.getURL("src/lib.js"));",
        "originalLength": 421,
        "removedExports": [],
        "renderedExports": [],
        "renderedLength": 306,
      },
      "vite-plugin-web-extension/test/manifest/resources/fullExtension/src/entries/contentScript/primary/style.css": {
        "code": "",
        "originalLength": 296,
        "removedExports": [],
        "renderedExports": [],
        "renderedLength": 0,
      },
      "vite-plugin-web-extension/test/manifest/resources/fullExtension/src/entries/contentScript/renderContent.js": {
        "code": "async function renderContent(
  cssPaths,
  render = (_appRoot) => {}
) {
  console.log("renderContent", cssPaths);
}",
        "originalLength": 133,
        "removedExports": [],
        "renderedExports": [
          "default",
        ],
        "renderedLength": 117,
      },
    },
    "name": "test/manifest/resources/fullExtension/src/entries/contentScript/primary/main",
    "referencedFiles": [],
    "type": "chunk",
    "viteMetadata": {
      "importedAssets": Set {},
      "importedCss": Set {
        "assets/main.css",
      },
    },
  },
  {
    "code": "import { l as logo } from "../../../../../../../logo.js";
const imageUrl = new URL(logo, import.meta.url).href;
document.querySelector("#app").innerHTML = \`
  <img src="\${imageUrl}" height="45" alt="" />
  <h1>Options</h1>
  <a href="https://vitejs.dev/guide/features.html" target="_blank">Vite Docs</a>
\`;
",
    "dynamicImports": [],
    "exports": [],
    "facadeModuleId": "vite-plugin-web-extension/test/manifest/resources/fullExtension/src/entries/options/index.html",
    "fileName": "assets/test/manifest/resources/fullExtension/src/entries/options/index.js",
    "implicitlyLoadedBefore": [],
    "importedBindings": {
      "assets/logo.js": [
        "l",
      ],
    },
    "imports": [
      "assets/logo.js",
    ],
    "isDynamicEntry": false,
    "isEntry": true,
    "isImplicitEntry": false,
    "map": null,
    "modules": {
      "vite-plugin-web-extension/test/manifest/resources/fullExtension/src/entries/options/index.html": {
        "code": "",
        "originalLength": 212,
        "removedExports": [],
        "renderedExports": [],
        "renderedLength": 0,
      },
      "vite-plugin-web-extension/test/manifest/resources/fullExtension/src/entries/options/main.js": {
        "code": "const imageUrl = new URL(logo, import.meta.url).href;

document.querySelector("#app").innerHTML = \`
  <img src="\${imageUrl}" height="45" alt="" />
  <h1>Options</h1>
  <a href="https://vitejs.dev/guide/features.html" target="_blank">Vite Docs</a>
\`;",
        "originalLength": 315,
        "removedExports": [],
        "renderedExports": [],
        "renderedLength": 249,
      },
      "vite-plugin-web-extension/test/manifest/resources/fullExtension/src/entries/options/style.css": {
        "code": "",
        "originalLength": 202,
        "removedExports": [],
        "renderedExports": [],
        "renderedLength": 0,
      },
    },
    "name": "test/manifest/resources/fullExtension/src/entries/options/index",
    "referencedFiles": [],
    "type": "chunk",
    "viteMetadata": {
      "importedAssets": Set {},
      "importedCss": Set {
        "assets/index.css",
      },
    },
  },
  {
    "fileName": "background.html",
    "name": undefined,
    "source": "<!DOCTYPE html><html lang="en"><head><meta charset="UTF-8" />  <script type="module" crossorigin src="/assets/background.js"></script>
</head></html>",
    "type": "asset",
  },
  {
    "fileName": "manifest.json",
    "name": undefined,
    "source": "{
  "version": "1.0.0",
  "name": "Manifest Name",
  "manifest_version": 2,
  "background": {
    "persistent": false,
    "page": "background.html"
  },
  "content_scripts": [
    {
      "js": [
        "test/manifest/resources/fullExtension/src/entries/contentScript/primary/main.js"
      ],
      "matches": [
        "*://*/*"
      ]
    }
  ],
  "browser_action": {
    "default_icon": {
      "32": "test/manifest/resources/fullExtension/src/assets/logo.svg"
    }
  },
  "options_ui": {
    "chrome_style": false,
    "open_in_tab": true,
    "page": "test/manifest/resources/fullExtension/src/entries/options/index.html"
  },
  "permissions": [
    "*://*/*"
  ],
  "web_accessible_resources": [
    "assets/logo.js",
    "assets/main.css",
    "assets/test/manifest/resources/fullExtension/src/entries/contentScript/primary/main.js",
    "test/manifest/resources/fullExtension/src/lib.js"
  ]
}",
    "type": "asset",
  },
  {
    "fileName": "test/manifest/resources/fullExtension/src/entries/contentScript/primary/main.js",
    "name": undefined,
    "source": "(async()=>{await import(chrome.runtime.getURL("assets/test/manifest/resources/fullExtension/src/entries/contentScript/primary/main.js"))})();",
    "type": "asset",
  },
  {
    "fileName": "test/manifest/resources/fullExtension/src/entries/options/index.html",
    "name": undefined,
    "source": "<!DOCTYPE html>
<html lang="en">
  <head>
    <meta charset="UTF-8" />
    <title>Options</title>
    <script type="module" crossorigin src="/assets/test/manifest/resources/fullExtension/src/entries/options/index.js"></script>
    <link rel="stylesheet" crossorigin href="/assets/index.css">
  </head>
  <body>
    <div id="app"></div>
  </body>
</html>
",
    "type": "asset",
  },
  {
    "fileName": "test/manifest/resources/fullExtension/src/lib.js",
    "name": undefined,
    "source": "console.log("lib.js");
",
    "type": "asset",
  },
]
`;
